examples/kubernetes/README.md

```diff
@@ -1,18 +1,124 @@
-Usage
+Ceph on Kubernetes
 =====
+This Guide will take you through the process of deploying a Ceph cluster on to a Kubernetes cluster.
 
-Create a file called `ceph-mon.json` with the repository content.
-Eventually run the pod like so:
+Sigil is required for template handling and must be installed in system PATH. Instructions can be found here: [https://github.com/gliderlabs/sigil](https://github.com/gliderlabs/sigil)
+# Generate keys and configuration
 
-`kubectl create -f mon.json`
+Run the following commands to generate the required configuration and keys.
 
-Check the status of the pod:
+```
+cd generator
+./generate_secrets.sh all `./generate_secrets.sh fsid`
+
+kubectl create namespace ceph
+
+kubectl create secret generic ceph-conf-combined --from-file=ceph.conf --from-file=ceph.client.admin.keyring --from-file=ceph.mon.keyring --namespace=ceph
+kubectl create secret generic ceph-bootstrap-rgw-keyring --from-file=ceph.keyring=ceph.rgw.keyring --namespace=ceph
+kubectl create secret generic ceph-bootstrap-mds-keyring --from-file=ceph.keyring=ceph.mds.keyring --namespace=ceph
+kubectl create secret generic ceph-bootstrap-osd-keyring --from-file=ceph.keyring=ceph.osd.keyring --namespace=ceph
+kubectl create secret generic ceph-client-key --from-file=ceph-client-key --namespace=ceph
+
+cd ..
+```
+
+Please note that you should save the output files of this command, they will overwrite existing keys and configuration. If you lose these files they can still be retrieved from Kubernetes via `kubectl get secret`.
+
+# Deploy Ceph Components
+
+With the secrets created, you can now deploy ceph.
+
+```
+kubectl create \
+-f ceph-mds-v1-rc.yaml \
+-f ceph-mon-v1-svc.yaml \
+-f ceph-mon-v1-ds.yaml \
+-f ceph-mon-check-v1-rc.yaml \
+-f ceph-osd-v1-ds.yaml \
+--namespace=ceph
+```
+
+Your cluster should now look something like this.
 
 ```
-$ kubectl get pod
-POD              IP            CONTAINER(S)   IMAGE(S)      HOST                  LABELS          STATUS    CREATED      MESSAGE
-ceph-mon-5ceys   172.17.0.13                                127.0.0.1/127.0.0.1   name=frontend   Running   49 minutes
-                               ceph-mon       ceph-daemon                                         Running   49 minutes
+$ kubectl get all --namespace=ceph
+NAME                   DESIRED      CURRENT       AGE
+ceph-mds               1            1             24s
+ceph-mon-check         1            1             24s
+NAME                   CLUSTER-IP   EXTERNAL-IP   PORT(S)    AGE
+ceph-mon               None         <none>        6789/TCP   24s
+NAME                   READY        STATUS        RESTARTS   AGE
+ceph-mds-6kz0n         0/1          Pending       0          24s
+ceph-mon-check-deek9   1/1          Running       0          24s
+```
+
+# Label your storage nodes
+
+You must label your storage nodes in order to run Ceph pods on them.
+
+```
+kubectl label node <nodename> node-type-storage
```

Comment:
Small typo on the label, it should be: `kubectl label node <nodename> node-type=storage`